src/PowerShellEditorServices/Session/PromptNestFrame.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerShell.EditorServices.Utility;
+
+namespace Microsoft.PowerShell.EditorServices.Session
+{
+    using System.Management.Automation;
```

Comment:
The style this code base has been using does not put using inside namespace decls.

Reply:
So when that using statement is outside of the namespace, any calls to the PowerShell API fail to compile with `'PowerShell' is a namespace but is used like a type [PowerShellEditorServices]`

Not sure what is causing that, I couldn't find a namespace that ends in `PowerShell` other than `Microsoft.PowerShell`.  And there's no using statement for `Microsoft`.

I put that using statement inside the namespace because that's how PowerShellContext is set up.  But I agree, I'm not particularly comfortable with that solution.  Anyone have any thoughts on how to better handle this?

Reply:
`using System.Management.Automation = SMA`?

Reply:
@rjmholt that would probably work, but would we prefer having `SMA.PowerShell` all over the place over having a using statement or two in the namespace?

I even tried `using PowerShell = System.Management.Automation.PowerShell` with no luck.  I think this is the lesser evil so to speak.